numpy/core/numeric.py

```diff
@@ -401,11 +402,17 @@ def asarray(a, dtype=None, order=None):
         Input data, in any form that can be converted to an array.  This
         includes lists, lists of tuples, tuples, tuples of tuples, tuples
         of lists and ndarrays.
-    dtype : data-type, optional
+    dtype : {None, data-type}, optional
         By default, the data-type is inferred from the input data.
-    order : {'C', 'F'}, optional
-        Whether to use row-major ('C') or column-major ('F' for FORTRAN)
-        memory representation.  Defaults to 'C'.
+    order : {'C', 'F', 'A'}, optional
+        Whether to use row-major ('C'), column-major ('F'), or any ('A')
+        memory representation.  Defaults to 'A'.
+    precision: {None, dtype, ndarray, scalar}, optional
```

Comment:
Nice that gives us 3-4 types of dtype logics, two of which may be usefully combined :). Note that result type actually does *not* use casting logic. Since it will always pick float64, unless it finds a lower precision float type (for floats, for ints it probably gets stranger with unsigned ints, but I am not concerned that much about those right now).